### PR TITLE
admin accessibility audit

### PR DESCRIPTION
**Pull request by TheDancingClown**
https://docs.google.com/spreadsheets/d/1dMAStRPPyQE4DU7YngCiUtvKEIHJDURqH2YmV9svXdE/edit#gid=0

### All user tables
- Removes email column
- Adds table header for edit column
- Adds aria to include names for edit links
- Adds button role and aria to clear search link/button
- Adds button tag to 'Add user' button

### Nomination table
- Adds button role and aria to clear filter link/button
- Adds aria to include names for view links

### Admins
- Removes signin notice - redundant as evident from context

### All users
- Removes signout notice - redundant as evident from context (on a login page)

### CODE REVIEW

File: app/views/admin/assessors/_list.html.slim [hunk 1]
```diff
@@ -54,4 +54,8 @@ div role="region" aria-labelledby="table-list-national-assessors-caption" tabind
                 .text-muted
                   ' Not locked
             td.govuk-table__cell
-              = link_to "Edit user", edit_admin_assessor_path(assessor), class: "govuk-link", id: "edit-#{ assessor.first_name.downcase }-#{ assessor.last_name.downcase }"
+              = link_to "Edit user",
+                        edit_admin_assessor_path(assessor),
+                        class: "govuk-link",
+                        id: "edit-#{ assessor.first_name.downcase }-#{ assessor.last_name.downcase }",
+                        aria: { label: "edit #{ assessor.full_name }"}
```
Comment (mauricioac) [inline app/views/admin/assessors/_list.html.slim:61]: Be careful of one thing: the visible text should ideally be contained inside the aria label.
Not sure what to do here snce they mismatch

File: app/views/admin/form_answers/_edit_nomination_links.html.slim [hunk 2]
```diff
@@ -1 +1,5 @@
-= link_to 'Edit nomination and local assessment form', [:edit, namespace_name, @form_answer], class: "govuk-button"
+= link_to 'Edit nomination and local assessment form',
+          [:edit, namespace_name, @form_answer],
+          class: "govuk-button",
+          role: "button",
+          aria: { label: "Edit nomination and local assessment form"}
```
Comment (mauricioac) [inline app/views/admin/form_answers/_edit_nomination_links.html.slim:5]: This aria label is the same as text, so can be removed

File: app/views/admin/form_answers/index.html.slim [hunk 3]
```diff
@@ -20,8 +20,16 @@ h1.govuk-heading-xl
...
+    = f.submit 'Apply filters',
+                class: 'govuk-button',
+                id: 'apply-nomination-filters',
+                aria: { label: "Search and apply filters" }
+                
+    = link_to "Remove filters",
+              admin_form_answers_path,
+              class: "govuk-button govuk-button--secondary",
+              role: "button",
+              aria: { label: "Remove filters" }
```
Comment (mauricioac) [inline app/views/admin/form_answers/index.html.slim:32]: remove: redundant aria label

File: app/views/admin/group_leaders/_list.html.slim [hunk 4]
```diff
@@ -27,8 +25,10 @@ div role="region" aria-labelledby="table-list-group-leaders-caption" tabindex="0
         - GroupLeaderDecorator.decorate_collection(resources).each do |group_leader|
           tr.govuk-table__row
             th.govuk-table__header scope="row"
-              = link_to group_leader.full_name, edit_admin_group_leader_path(group_leader), class: "govuk-link"
-            td.govuk-table__cell = mail_to group_leader.email, group_leader.email, {class: "ellipsis govuk-link"}
+              = link_to group_leader.full_name,
+                        edit_admin_group_leader_path(group_leader),
+                        class: "govuk-link",
+                        aria: { label: "edit-#{ group_leader.first_name.downcase }-#{ group_leader.last_name.downcase }" }
```
Comment (mauricioac) [inline app/views/admin/group_leaders/_list.html.slim:31]: why is this aria label formatted like an ID?

File: app/views/admin/group_leaders/_list.html.slim [hunk 5]
```diff
@@ -51,4 +51,8 @@ div role="region" aria-labelledby="table-list-group-leaders-caption" tabindex="0
                 .text-muted
                   ' Not locked
             td.govuk-table__cell
-              = link_to "Edit user", edit_admin_group_leader_path(group_leader), class: "govuk-link", id: "edit-#{ group_leader.first_name.downcase }-#{ group_leader.last_name.downcase }"
+              = link_to "Edit user",
+                edit_admin_group_leader_path(group_leader),
+                class: "govuk-link",
+                id: "edit-#{ group_leader.first_name.downcase }-#{ group_leader.last_name.downcase }",
+                aria: { label: "edit-#{ group_leader.first_name.downcase }-#{ group_leader.last_name.downcase }" }
```
Comment (mauricioac) [inline app/views/admin/group_leaders/_list.html.slim:58]: same here

File: app/views/admin/group_leaders/index.html.slim [hunk 6]
```diff
@@ -10,7 +10,11 @@
               = f.input :query, label: "Search", input_html: { class: "form-control", type: "search" }
             .govuk-button-group
               = f.submit "Search", class: 'govuk-button'
-              = link_to "Clear search", [:admin, controller_name.to_sym], class: "govuk-button govuk-button--secondary"
+              = link_to "Clear search",
+                        [:admin, controller_name.to_sym],
+                        class: "govuk-button govuk-button--secondary",
+                        role: "button",
+                        aria: { label: "Clear search" }
```
Comment (mauricioac) [inline app/views/admin/group_leaders/index.html.slim:17]: redundant aria label, remove

File: app/views/admin/lieutenants/_list.html.slim [hunk 7]
```diff
@@ -31,8 +29,10 @@ div role="region" aria-labelledby="table-list-lieutenancy-office-caption" tabind
         - LieutenantDecorator.decorate_collection(resources).each do |lieutenant|
           tr.govuk-table__row
             th.govuk-table__header scope="row"
-              = link_to lieutenant.full_name, edit_admin_lieutenant_path(lieutenant), class: "govuk-link"
-            td.govuk-table__cell = mail_to lieutenant.email, lieutenant.email, {class: "ellipsis govuk-link"}
+              = link_to lieutenant.full_name,
+                edit_admin_lieutenant_path(lieutenant),
+                class: "govuk-link",
+                aria: { label: "edit-#{ lieutenant.first_name.downcase }-#{ lieutenant.last_name.downcase }" }
```
Comment (mauricioac) [inline app/views/admin/lieutenants/_list.html.slim:35]: formatted as ID again

File: app/views/admin/lieutenants/_list.html.slim [hunk 8]
```diff
@@ -59,4 +59,7 @@ div role="region" aria-labelledby="table-list-lieutenancy-office-caption" tabind
                 .text-muted
                   ' Not locked
             td.govuk-table__cell
-              = link_to "Edit user", edit_admin_lieutenant_path(lieutenant), class: "govuk-link", id: "edit-#{ lieutenant.first_name.downcase }-#{ lieutenant.last_name.downcase }"
+              = link_to "Edit user", edit_admin_lieutenant_path(lieutenant),
+                        class: "govuk-link",
+                        id: "edit-#{ lieutenant.first_name.downcase }-#{ lieutenant.last_name.downcase }",
+                        aria: { label: "edit-#{ lieutenant.first_name.downcase }-#{ lieutenant.last_name.downcase }" }
```
Comment (mauricioac) [inline app/views/admin/lieutenants/_list.html.slim:65]: same again

File: app/views/admin/users/index.html.slim [hunk 9]
```diff
@@ -10,11 +10,15 @@
               = f.input :query, label: "Search", input_html: { class: "form-control", type: "search" }
         .govuk-button-group
           = f.submit "Search", class: 'govuk-button'
-          = link_to "Clear search", [:admin, controller_name.to_sym], class: "govuk-button govuk-button--secondary"
+          = link_to "Clear search",
+                    [:admin, controller_name.to_sym],
+                    class: "govuk-button govuk-button--secondary",
+                    role: "button",
+                    aria: { label: "Clear search" }
```
Comment (mauricioac) [inline app/views/admin/users/index.html.slim:17]: redundant, remove

File: app/views/assessor/form_answers/index.html.slim [hunk 10]
```diff
@@ -12,11 +12,17 @@ h1.govuk-heading-xl
...
 
-  = hidden_field_tag "search[sort]", params.dig(:search, :sort)
-
   .govuk-button-group
-    = f.submit 'Search and apply filters', class: 'govuk-button', id: 'apply-nomination-filters'
-    = link_to "Clear search and filters", assessor_form_answers_path, class: "govuk-button govuk-button--secondary"
+    = f.submit 'Search and apply filters',
+                class: 'govuk-button',
+                id: 'apply-nomination-filters',
+                aria: { label: "Search and apply filters" }
```
Comment (mauricioac) [inline app/views/assessor/form_answers/index.html.slim:19]: redundant, remove

File: app/views/assessor/form_answers/index.html.slim [hunk 11]
```diff
@@ -12,11 +12,17 @@ h1.govuk-heading-xl
...
+                
+    = link_to "Clear search and filters",
+              assessor_form_answers_path,
+              class: "govuk-button govuk-button--secondary",
+              role: "button",
+              aria: { label: "Remove filters" }
```
Comment (mauricioac) [inline app/views/assessor/form_answers/index.html.slim:25]: Be careful as this aria label is very different from the actual text. I would just leave the visible text, there's nothing wrong with it that this ARIA label would fix

File: app/views/lieutenant/form_answers/index.html.slim [hunk 12]
```diff
@@ -11,11 +11,17 @@ h1.govuk-heading-xl
...
 
-  = hidden_field_tag "search[sort]", params.dig(:search, :sort)
-
   .govuk-button-group
-    = f.submit 'Search and apply filters', class: 'govuk-button', id: 'apply-nomination-filters'
-    = link_to "Clear search and filters", lieutenant_form_answers_path, class: "govuk-button govuk-button--secondary"
+    = f.submit 'Search and apply filters',
+                class: 'govuk-button',
+                id: 'apply-nomination-filters',
+                aria: { label: "Search and apply filters" }
```
Comment (mauricioac) [inline app/views/lieutenant/form_answers/index.html.slim:18]: redundant, remove

File: app/views/lieutenant/form_answers/index.html.slim [hunk 13]
```diff
@@ -11,11 +11,17 @@ h1.govuk-heading-xl
...
+
+    = link_to "Clear search and filters",
+              lieutenant_form_answers_path,
+              class: "govuk-button govuk-button--secondary",
+              role: "button",
+              aria: { label: "Clear search filters" }
```
Comment (mauricioac) [inline app/views/lieutenant/form_answers/index.html.slim:24]: I would just leave the visible text. This aria label mismatches from visible text and has different semantics, which would be more trouble in an audit.

File: app/views/lieutenant/lieutenants/index.html.slim [hunk 14]
```diff
@@ -8,12 +8,20 @@
           = f.input :query, label: "Search", input_html: { class: "form-control govuk-!-width-one-half", type: "search" }
     .govuk-button-group
       = f.submit "Search", class: 'govuk-button'
-      = link_to "Clear search", lieutenant_lieutenants_path, class: "govuk-button govuk-button--secondary"
+      = link_to "Clear search",
+                lieutenant_lieutenants_path,
+                class: "govuk-button govuk-button--secondary",
+                role: "button",
+                aria: { label: "Clear search" }
```
Comment (mauricioac) [inline app/views/lieutenant/lieutenants/index.html.slim:15]: redundant, remove

File: app/views/admin/assessors/_list.html.slim (hunk 1 above)
Comment (TheDancingClown) [reply app/views/admin/assessors/_list.html.slim:61]: Will flag with UI as visual reference is heavily relied upon for context - but aria label is adequate to show purpose of link

**Pull request merged**